controllers/vrg_volrep.go

```diff
@@ -512,9 +512,40 @@ func (v *VRGInstance) getPVFromPVC(pvc *corev1.PersistentVolumeClaim) (corev1.Pe
 	return pv, nil
 }
 
-func (v *VRGInstance) getObjectStore(s3ProfileName string) (ObjectStorer, error) {
-	return v.reconciler.ObjStoreGetter.ObjectStore(v.ctx, v.reconciler.APIReader,
-		s3ProfileName, v.namespacedName, v.log)
+func (v *VRGInstance) getObjectStorer(s3ProfileName string) (ObjectStorer, error) {
+	objectStore, err := v.getCachedObjectStorer(s3ProfileName)
+	if objectStore != nil || err != nil {
+		return objectStore, err
+	}
```

Comment:
(nit) I suggest replacing these lines with the content of `getCachedObjectStorer` (lines 537-539) and deleting it to avoid testing two additional conditions on line 517